mern-ecommerce/controllers/user.js

```diff
@@ -125,3 +125,19 @@ exports.purchaseHistory = (req, res) => {
       res.json(orders);
     });
 };
+
+exports.deleteUser = (req, res) => {
+  
+  const { userId } = req.params;
+  User.findByIdAndRemove(userId, (err, user) => {
+    if (err || !user) {
+      return res.status(400).json({
```

Comment:
The code snippet is returning a 400 status code with an error message. This is a common pattern in the repository context. However, it would be better to have a centralized error handling mechanism instead of repeating this code throughout the application. Consider creating a separate function or middleware to handle errors in a consistent manner.

Additionally, the `errorHandler` function is being used to handle errors, but its implementation is not provided in the given context. It's essential to ensure that this function is correctly handling errors and providing useful error messages.